concent_api/common/decorators.py

```diff
@@ -85,3 +89,32 @@ def decorated(*args: Any, **kwargs: Any) -> None:
                 return func(*args, **kwargs)
 
         return decorated
+
+
+class retry_on_database_error(object):
+
+    def __init__(self, func: Callable) -> None:
+        self.func = func
+        self.max_number_of_retries = settings.MAXIMUM_NUMBER_OF_RETRIES  # pylint: disable=attribute-defined-outside-init
+
+    def __call__(self, *args: Any, **kwargs: Any) -> Any:
+        self._initialize_retry_parameters()
+        while True:
+            try:
+                result = self.func(*args, **kwargs)
+                break
+            except IntegrityError:
+                self._validate_number_of_retries()
```

Comment:
Shouldn't you also check here if the exception type is `pg_errorcodes.UNIQUE_VIOLATION`?